### DISCUSSION TITLE
IntelliSense for Cypress library

### DISCUSSION THREAD

**Discussion by NormalCrazy000** [cypress-io/cypress#30068] [Questions and Help] [answered]
Reading through the documentation, I can't understand why it is necessary to put the reference to the Cypress library inside jsconfig.json/tsconfig.json to use IntelliSense on VSCode(in addition to the Triple slash directives mode) and it is not possible to use package.json. Is there any trick that allows this?
Thanks

**Comment by jennifer-shehane** [comment, marked as answer]
IntelliSense in VS Code is driven by the TypeScript language service, which reads `tsconfig.json`/`jsconfig.json` — not your project’s `package.json`. There’s no `package.json` field the language service consults to inject global types, so there’s no trick to move it there.

The confusion is that the types field does matter, but only for a package’s own declarations: `/// <reference types="cypress" />` resolves `node_modules/cypress/package.json` → `types/index.d.ts`. Your project’s `package.json` can’t register those Cypress globals (cy, Cypress, expect) for the editor.

The three supported ways (all editor-side):

```js
// 1. per-file
/// <reference types="cypress" />
```

```json
// 2. jsconfig.json (JS projects)
{ "include": ["./node_modules/cypress", "cypress/**/*.js"] }
```

```json
// 3. tsconfig.json (TS projects)
{ "compilerOptions": { "allowJs": true, "types": ["cypress"] }, "include": ["**/*.*"] }
```

For project-wide coverage without per-file directives, use `"types": ["cypress"]` in tsconfig.json/jsconfig.json.
